sources/Context.cpp

```diff
@@ -17,6 +17,29 @@
 
 #include "Context.h"
 
+#include <tuple>
+
 namespace dfl {
-Context::Context(const std::string& networkFilepath) : networkManager_(networkFilepath) {}
+Context::Context(const std::string& networkFilepath) : networkManager_(networkFilepath) {
+  networkManager_.onNode(std::bind(&Context::processSlackNode, this, std::placeholders::_1));
+}
+
+void
+Context::processSlackNode(const boost::shared_ptr<inputs::Node>& node) {
+  if (!slack_node_) {
```

Comment:
I think it would be good to also write unit tests for those algorithms

Reply:
Solution: Algorithms will be written in a new library that will be unit-tested, and Context will simply aggregate all the algorithms from the library. Will be done in a future PR.

Reply:
Algorithms will be put in a separated ibrary to be unit-tested